src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -39,6 +39,14 @@ internal sealed class MsQuicConnection : QuicConnectionProvider
         private bool _remoteCertificateRequired;
         private X509RevocationMode _revocationMode = X509RevocationMode.Offline;
         private RemoteCertificateValidationCallback? _remoteCertificateValidationCallback;
+        private string? _targetHost;
+
+        private string TargetHost => _targetHost ?? (_targetHost = _remoteEndPoint switch
```

Comment:
@wfurt I needed the same thing as you in #55468. We'll clash on this and one of us will have to fix their part.